agent/structs/config_entry_status.go

```diff
@@ -31,11 +45,66 @@ type Status struct {
 	Conditions []Condition
 }
 
+func (s Status) SameConditions(other Status) bool {
```

Comment:
This is pretty much only used in our testing code for equality checking the conditions that we're expecting to be set on an object and the actual conditions set, so I may move it there.